astroid/nodes/node_classes.py

```diff
@@ -1202,62 +1202,25 @@ class Assign(_base_nodes.AssignTypeNode, _base_nodes.Statement):
     <Assign l.1 at 0x7effe1db8550>
     """
 
-    _astroid_fields = ("targets", "value")
-    _other_other_fields = ("type_annotation",)
-
-    def __init__(
-        self,
-        lineno: int | None = None,
-        col_offset: int | None = None,
-        parent: NodeNG | None = None,
-        *,
-        end_lineno: int | None = None,
-        end_col_offset: int | None = None,
-    ) -> None:
-        """
-        :param lineno: The line that this node appears on in the source code.
-
-        :param col_offset: The column that this node appears on in the
-            source code.
-
-        :param parent: The parent node in the syntax tree.
-
-        :param end_lineno: The last line this node appears on in the source code.
-
-        :param end_col_offset: The end column this node appears on in the
-            source code. Note: This is after the last symbol.
-        """
-        self.targets: list[NodeNG] = []
-        """What is being assigned to."""
+    targets: list[NodeNG]
+    """What is being assigned to."""
 
-        self.value: NodeNG | None = None
-        """The value being assigned to the variables."""
+    value: NodeNG
+    """The value being assigned to the variables."""
 
-        self.type_annotation: NodeNG | None = None  # can be None
-        """If present, this will contain the type annotation passed by a type comment"""
+    type_annotation: NodeNG | None
+    """If present, this will contain the type annotation passed by a type comment"""
 
-        super().__init__(
-            lineno=lineno,
-            col_offset=col_offset,
-            end_lineno=end_lineno,
-            end_col_offset=end_col_offset,
-            parent=parent,
-        )
+    _astroid_fields = ("targets", "value")
+    _other_other_fields = ("type_annotation",)
 
     def postinit(
```

Comment:
https://github.com/pylint-dev/astroid/blob/345b91cd6d29318ac34352699ff75a0132ccda15/astroid/rebuilder.py#L808

Could you add asserts here that the typing we add here is correct?

And could you add a changelog entry? For the rest this LGTM!